spec/lib/response-handler.spec.js

```diff
@@ -0,0 +1,128 @@
+        });
+        it("returns true if the status code is greater than 399", () => {
+            expect(responseHandler.isErrorCode(404)).toBe(true);
+        });
+    });
+    describe(".createErrorForResponse()", () => {
+        var errorResponse;
+
+        it("sets the resposne", () => {
+            errorResponse = {
+                error: "ExampleError",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "Example Error Message"
+                    }
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("Example Error Message"));
+        });
+        it("returns any unkown errors", () => {
+            errorResponse = {
+                body: {
+                    code: "anUnkownError"
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("UNKOWN", "UNKOWN"));
+        });
+        it("returns a timeout error", () => {
+            errorResponse = {
+                code: "exampleTimeoutError"
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("ESOCKETTIMEDOUT"));
+        });
+        it("returns known errors", () => {
+            errorResponse = {
+                error: "duplicate_artifact",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "A duplicate artifact error message"
+                    }
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("A duplicate artifact error message"));
+        });
+    });
+    describe(".handleErrorResponse()", () => {
+        it("throws errors from responseHandler.createErrorForResponse()", () => {
+            spyOn(responseHandler, "createErrorForResponse").andReturn(new Error("SomeError"));
+            expect(() => {
+                responseHandler.handleErrorResponse();
+            }).toThrow();
+        });
+    });
+    describe(".resolveLink()", () => {
+        it("returns a new URI as a string", () => {
+            var link, resolvedLink, url;
+
+            link = "link/example";
+            url = "url/example";
+            resolvedLink = responseHandler.resolveLink(link, url);
+            expect(resolvedLink).toBe("link/url/example");
+        });
+    });
+    describe(".resolveLinkHeaders()", () => {
+        it("returns an array of urls", () => {
+            var response, urls;
+
+            response = {
+                headers: {
+                    link: "<https://api.github.com/user/9287/repos?page=3&per_page=100>; rel=\"item\", <https://api.github.com/user/9287/repos?page=1&per_page=100>; rel=\"item\"; pet='cat', <https://api.github.com/user/9287/repos?page=5&per_page=100>; rel=\"item\""
```

Comment:
I actually had noticed (I think) that request does indeed condense them all as a single link header.